### ISSUE TITLE
Formatting lost after build is finished

### ISSUE THREAD

**Issue by vercas** [randy3k/Terminus#108]
When using Terminus for builds, I can see colored output just fine from the build command (e.g. from GCC) while the build is running, but it's all gone when the build is finished and the text remains plain white.

**Comment by randy3k** [comment]
could you do some screenshots or provide the build configuration?

**Comment by vercas** [comment]
Well, yes, but it seems to be a bit more complicated than what I first explained.  
I do the build, I see colored output just fine, then it finishes, and I still see colors and stuff...  
Until my buffer/window/whatever suddenly scrolls to the bottom and all the formatting is gone.  
After writing all this text here, I went back to Sublime and now the colors are back again!  
Now I pressed the Escape key, it took me to the bottom, and colors are gone again... I alt-tabbed back into Sublime after writing this sentence and the colors are back...

This is my build config: https://github.com/vercas/Beelzebub/blob/master/Beelzebub.sublime-project#L63
And I'm attaching a (mostly useless AFAICT) screenshot of colored output whilst building
![snapshot40](https://user-images.githubusercontent.com/1823741/56227777-c4ba0700-6065-11e9-942f-54aa39e25a9a.png)

**Comment by randy3k** [comment]
That’s odd. (Ya, you are right. That screenshot is useless). Could you turn on the debug settings in Terminus preferences then repeat the `Escape` thing? Report what you see in the sublime console after the Escape key is hit.

**Comment by macowie** [comment]
I've seen this a couple times using `"target": "toggle_terminus_panel"` for my RSpec build task. I want to say it's usually after the process exits, I haven't a sense of a pattern yet. Rejiggering the window does make the colors come back.

**Comment by randy3k** [comment]
With this build system, I always get a colored output

```
	"build_systems":
	[
		{
			"auto_close": false,
			"cmd":
			[
				"bash",
				"-c",
				"echo -e \"\\033[31mHello World\""
			],
			"name": "Hello World",
			"target": "terminus_open",
			"working_dir": "$folder"
		}
	],
```

**Comment by randy3k** [comment]
@macowie

`toggle_terminus_panel` should not be used as a build target, please use `terminus_open` instead. You just have to specify `panel_name` if you want to open the terminal in a panel,

**Comment by macowie** [comment]
Yes, I had figured that out after I posted. Thanks.

In my case, something is rerunning the build. Focusing or unfocusing the panel seems to trigger it? I can see the process spin up afterwards. The text output doesn't change but the coloring is overwritten. First all white then the new color is overlaid (incorrectly, as the test order is randomised) as the command runs.

![image](https://user-images.githubusercontent.com/2200578/56245381-e666be80-6064-11e9-9421-2698f6e3ca26.png)

I guess this two things, do I need to clear the panel myself before a build? And, what would cause the command to rerun?


This is my current build command
```
{
  "title": "RSpec",
	"cmd": ["bundle", "exec", "rspec"],
	"working_dir": "${project_path:${folder}}",
	"selector": "source.ruby.rspec",
	"target": "terminus_open",
  "panel_name": "RSpec",
  "auto_close": false,
	"variants": [
	    {
	      "name": "Current File",
				"cmd": ["bundle", "exec", "rspec", "$file"],
	    }
	]
}
```

I only have a few packages installed, Terminus, Origami, and LSP.

**Comment by randy3k** [comment]
It will be great if someone can come up with a small reproducible example.

**Comment by vercas** [comment]
I think @macowie describes the problem - something is re-running the build.  
I noticed that build artifacts are missing sometimes when building with Terminus.  
![image](https://user-images.githubusercontent.com/1823741/56251532-da521f80-60ab-11e9-9249-289037d094d6.png)  
You can see the odd behaviour... I did a build before each file listing except the one where the directory vanished.  
The only reason I can think of why my directory vanished is because the build restarted spuriously - deleting that directory is the first step in the (full) build process.

**Comment by vercas** [comment]
Yeah, I can confirm that it does spurious builds.  
While typing the previous comment, it decided to rebuild everything.  
The reason why only some of the files are missing in some listings in the previous screenshot is because those files take the longest to produce (`beelzebub.bin` is ~200 object files built with LTO).

**Comment by randy3k** [comment]
I think I know what's happening now. There is a feature to relaunch the terminal if the process is terminated. The same mechanism must have been triggered in your cases. Obviously, it is only useful for regular shell and is not desired for a build system.

**Comment by randy3k** [comment]
I have pushed a fix and made a release v0.2.30. It will be available from package control soon.

**Comment by vercas** [comment]
Can't reproduce it right now. Will keep using it today to see if I encounter the issue anymore.

**Comment by macowie** [comment]
Yes, this fixed it for me. Running a build is working as expected. I also opened a regular shell in a panel and tried various things and it all worked just fine. Thanks!
